helidon mp config does not work on windows

helidon-quickstart-mp fails on windows:

```
Exception in thread "main" io.helidon.config.ConfigException: Cannot load data from mandatory source UrlConfig[jar:file:/D:/fusion/darwin/helidonmp/helidon-quickstart-mp/target/helidon-quickstart-mp.jar!/META-INF/microprofile-config.properties]. Configuration at url 'jar:file:/D:/fusion/darwin/helidonmp/helidon-quickstart-mp/target/helidon-quickstart-mp.jar!/META-INF/microprofile-config.properties' GET is not accessible.
       at io.helidon.config.spi.AbstractSource.processMissingData(AbstractSource.java:241)
       at io.helidon.config.spi.AbstractSource.loadDataChangedSinceLastLoad(AbstractSource.java:208)
       at io.helidon.config.spi.AbstractSource.load(AbstractSource.java:177)
       at io.helidon.config.CompositeConfigSource.load(CompositeConfigSource.java:123)
       at io.helidon.config.ProviderImpl.newConfig(ProviderImpl.java:106)
       at io.helidon.config.BuilderImpl.build(BuilderImpl.java:254)
       at io.helidon.microprofile.config.MpConfigBuilder.build(MpConfigBuilder.java:244)
       at io.helidon.microprofile.config.MpConfigProviderResolver.buildConfig(MpConfigProviderResolver.java:73)
       at io.helidon.microprofile.config.MpConfigProviderResolver.getConfig(MpConfigProviderResolver.java:57)
       at io.helidon.microprofile.server.Server$Builder.doBuild(Server.java:211)
       at io.helidon.common.context.Contexts.runInContext(Contexts.java:118)
       at io.helidon.microprofile.server.Server$Builder.build(Server.java:202)
       at io.helidon.microprofile.server.Server.create(Server.java:94)
       at io.helidon.examples.quickstart.mp.Main.startServer(Main.java:58)
       at io.helidon.examples.quickstart.mp.Main.main(Main.java:45)
Caused by: io.helidon.config.ConfigException: Configuration at url 'jar:file:/D:/fusion/darwin/helidonmp/helidon-quickstart-mp/target/helidon-quickstart-mp.jar!/META-INF/microprofile-config.properties' GET is not accessible.
       at io.helidon.config.internal.UrlConfigSource.content(UrlConfigSource.java:107)
       at io.helidon.config.spi.AbstractParsableConfigSource.loadData(AbstractParsableConfigSource.java:58)
       at io.helidon.config.RetryPolicies$JustCallRetryPolicyHolder$1.execute(RetryPolicies.java:77)
       at io.helidon.config.spi.AbstractSource.loadDataChangedSinceLastLoad(AbstractSource.java:197)
       ... 13 more
Caused by: java.nio.file.InvalidPathException: Illegal char <:> at index 4: file:/D:/fusion/darwin/helidonmp/helidon-quickstart-mp/target/helidon-quickstart-mp.jar!/META-INF/microprofile-config.properties
       at sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
       at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
       at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
       at sun.nio.fs.WindowsPath.parse(WindowsPath.java:94)
       at sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:255)
       at java.nio.file.Paths.get(Paths.java:84)
       at io.helidon.config.internal.UrlConfigSource.probeContentType(UrlConfigSource.java:160)
       at io.helidon.config.internal.UrlConfigSource.mediaType(UrlConfigSource.java:149)
       at io.helidon.config.internal.UrlConfigSource.genericContent(UrlConfigSource.java:112)
       at io.helidon.config.internal.UrlConfigSource.content(UrlConfigSource.java:102)
       ... 16 more
```

This issue does not occur on 1.2.1

--- a/config/config/src/main/java/io/helidon/config/internal/UrlConfigSource.java
+++ b/config/config/src/main/java/io/helidon/config/internal/UrlConfigSource.java
@@ -165,7 +165,7 @@ public class UrlConfigSource extends AbstractParsableConfigSource<Instant> {
             case "jar":
                 String relativePath = uri.getSchemeSpecificPart();
                 int idx = relativePath.indexOf("!");
-                if (idx > 0) {
+                if (idx > 0 && idx < relativePath.length()) {
                     relativePath = relativePath.substring(idx + 1);
                 }
                 path = Paths.get(relativePath);

--- a/config/config/src/main/java/io/helidon/config/internal/UrlConfigSource.java
+++ b/config/config/src/main/java/io/helidon/config/internal/UrlConfigSource.java
@@ -20,9 +20,12 @@ import java.io.IOException;
 import java.io.InputStreamReader;
 import java.io.Reader;
 import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.net.URL;
 import java.net.URLConnection;
 import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.time.Instant;
 import java.util.Objects;
@@ -108,7 +111,7 @@ public class UrlConfigSource extends AbstractParsableConfigSource<Instant> {
         }
     }
 
-    private ConfigParser.Content<Instant> genericContent(URLConnection urlConnection) throws IOException {
+    private ConfigParser.Content<Instant> genericContent(URLConnection urlConnection) throws IOException, URISyntaxException {
         final String mediaType = mediaType(null);
         Reader reader = new InputStreamReader(urlConnection.getInputStream(),
                                               StandardCharsets.UTF_8);
@@ -116,7 +119,7 @@ public class UrlConfigSource extends AbstractParsableConfigSource<Instant> {
         return ConfigParser.Content.create(reader, mediaType, Optional.of(Instant.now()));
     }
 
-    private ConfigParser.Content<Instant> httpContent(HttpURLConnection connection) throws IOException {
+    private ConfigParser.Content<Instant> httpContent(HttpURLConnection connection) throws IOException, URISyntaxException {
         connection.setRequestMethod(GET_METHOD);
 
         final String mediaType = mediaType(connection.getContentType());
@@ -141,7 +144,7 @@ public class UrlConfigSource extends AbstractParsableConfigSource<Instant> {
         return super.mediaType();
     }
 
-    private String mediaType(String responseMediaType) {
+    private String mediaType(String responseMediaType) throws URISyntaxException {
         String mediaType = mediaType();
         if (mediaType == null) {
             mediaType = responseMediaType;
@@ -155,9 +158,25 @@ public class UrlConfigSource extends AbstractParsableConfigSource<Instant> {
         return mediaType;
     }
 
-    private String probeContentType() {
-        String path = url.getPath();
-        return ConfigHelper.detectContentType(Paths.get(path));
+    private String probeContentType() throws URISyntaxException {
+        URI uri = url.toURI();
+        Path path;
+        switch (uri.getScheme()) {
+            case "jar":
+                String relativePath = uri.getSchemeSpecificPart();
+                int idx = relativePath.indexOf("!");
+                if (idx > 0) {
+                    relativePath = relativePath.substring(idx + 1);
+                }
+                path = Paths.get(relativePath);
+                break;
+            case "file":
+                path = Paths.get(uri);
+                break;
+            default:
+                path = Paths.get(url.getPath());
+        }
+        return ConfigHelper.detectContentType(path);
     }
 
     @Override

